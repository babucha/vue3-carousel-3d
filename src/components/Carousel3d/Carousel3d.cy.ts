import Carousel3d from './Carousel3d.vue'
import Slide from '../Slide/Slide.vue'

import { h } from 'vue'

describe('<Carousel3d />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-vue
    cy.mount(Carousel3d, {
      slots: {
        default: () => [
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {})),
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {})),
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {}))
        ]
      }
    })
  })
})
