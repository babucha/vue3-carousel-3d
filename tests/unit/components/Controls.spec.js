import { mount } from '@vue/test-utils'

import Carousel3d from '../../../src/components/Carousel3d.vue'

import { getNSlides } from '../utils'

describe('Controls', () => {
  let wrapper
  let $controls

  beforeEach(async () => {
    wrapper = mount(Carousel3d, {
      props: {
        controlsVisible: true
      },
      slots: {
        default: () => getNSlides(4, true)
      }
    })
    await wrapper.vm.$nextTick()

    $controls = wrapper.vm.$el.querySelector('.carousel-3d-controls')
  })

  it('should mount successfully', () => {
    expect($controls).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a next button', () => {
    expect(wrapper.vm.$el.querySelector('.next')).toBeDefined()
  })

  it('should render a prev button', () => {
    expect(wrapper.vm.$el.querySelector('.prev')).toBeDefined()
  })

  it('should trigger onNext when next is clicked', () => {
    wrapper.vm.$el.querySelector('.next').click()

    return wrapper.vm.$nextTick().then(() => {
      expect(wrapper.vm.currentIndex).toBe(1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('should trigger onNext when next is clicked', () => {
    wrapper.vm.$el.querySelector('.prev').click()

    return wrapper.vm.$nextTick().then(() => {
      expect(wrapper.vm.currentIndex).toBe(3)
      expect(wrapper).toMatchSnapshot()
    })
  })

  // TODO: one directional tests
})
