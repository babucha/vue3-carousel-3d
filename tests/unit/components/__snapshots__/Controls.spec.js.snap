// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Controls > should mount successfully 1`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide current" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;">
      <div data-index="0" data-is-current="true" data-left-index="-1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide right-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(240px) translateZ(-400px) rotateY(-35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="1" data-left-index="-1" data-right-index="0"></div>
    </div>
    <div class="carousel-3d-slide left-2" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-480px) translateZ(-500px) rotateY(35deg); top: 0px; z-index: 997; opacity: 1; visibility: visible;">
      <div data-index="2" data-left-index="1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide left-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-240px) translateZ(-400px) rotateY(35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="3" data-left-index="0" data-right-index="-1"></div>
    </div>
  </div>
  <div class="carousel-3d-controls">
    <a aria-label="Previous slide" class="prev" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>‹</span>
    </a>
    <a aria-label="Next slide" class="next" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>›</span>
    </a>
  </div>
</div>
`;

exports[`Controls > should trigger onNext when next is clicked 1`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide left-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-240px) translateZ(-400px) rotateY(35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="0" data-left-index="0" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide current" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;">
      <div data-index="1" data-is-current="true" data-left-index="-1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide right-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(240px) translateZ(-400px) rotateY(-35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="2" data-left-index="-1" data-right-index="0"></div>
    </div>
    <div class="carousel-3d-slide left-2" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-480px) translateZ(-500px) rotateY(35deg); top: 0px; z-index: 997; opacity: 1; visibility: visible;">
      <div data-index="3" data-left-index="1" data-right-index="-1"></div>
    </div>
  </div>
  <div class="carousel-3d-controls">
    <a aria-label="Previous slide" class="prev" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>‹</span>
    </a>
    <a aria-label="Next slide" class="next" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>›</span>
    </a>
  </div>
</div>
`;

exports[`Controls > should trigger onNext when next is clicked 2`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide right-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(240px) translateZ(-400px) rotateY(-35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="0" data-left-index="-1" data-right-index="0"></div>
    </div>
    <div class="carousel-3d-slide left-2" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-480px) translateZ(-500px) rotateY(35deg); top: 0px; z-index: 997; opacity: 1; visibility: visible;">
      <div data-index="1" data-left-index="1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide left-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-240px) translateZ(-400px) rotateY(35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="2" data-left-index="0" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide current" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;">
      <div data-index="3" data-is-current="true" data-left-index="-1" data-right-index="-1"></div>
    </div>
  </div>
  <div class="carousel-3d-controls">
    <a aria-label="Previous slide" class="prev" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>‹</span>
    </a>
    <a aria-label="Next slide" class="next" href="#" style="width: 50px; height: 50px; line-height: 50px;">
      <span>›</span>
    </a>
  </div>
</div>
`;
